docs/spec_draft.md

```diff
@@ -300,6 +316,148 @@ the IEEE-754 specification. For boolean element type, the behavior is same as
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_gather
+
+### Semantics
+
+For each replica sub-group in `replica_groups`, the operation
```

Comment:
Can you rename all `sub-group` to `subgroup` here and in the ODS? I think it's okay to omit the dash just like `subset`.